src/ast/callback_visitor.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "ast.h"
+#include <functional>
+
+namespace bpftrace {
+namespace ast {
+
+using callback = std::function<void(Node *)>;
+
+class CallbackVisitor : public Visitor
+{
+public:
+  CallbackVisitor(callback func) : func_(func)
```

Comment:
Actually, now I think you'll want the `explicit` :p

Reply:
oops!